From 1c3b70e7d8499420c64a13468533d8928cae16ba Mon Sep 17 00:00:00 2001
From: Stephan Reichholf <reichi@opendreambox.org>
Date: Fri, 3 Mar 2017 12:40:54 +0100
Subject: [PATCH 1/2] Add some Pseudo SNR-Calculation (it actually is the
 average of level and quality), using only quality doesn't seem right to me

---
 rtp.cpp    | 2 +-
 rtp.h      | 4 ++--
 vtuner.cpp | 4 ++--
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/rtp.cpp b/rtp.cpp
index a80f1ae..95cd3d4 100644
--- a/rtp.cpp
+++ b/rtp.cpp
@@ -165,7 +165,7 @@ void satipRTP::parseRtcpAppPayload(char *buffer)
 		strp = strstr(strp, ",");
 		sscanf(strp, ",%d,%d,%d,%*s", &level, &lock, &quality);
 
-		m_signalStrength = (level >= 0) ? (level * 65535 / 255) : 0x12345678;
+		m_signalStrength = (level >= 0) ? (level * 65535 / 255) : 0;
 		m_hasLock = !!lock;
 		m_signalQuality = (m_hasLock && (quality >= 0)) ? (quality * 65535 / 15) : 0;
 
diff --git a/rtp.h b/rtp.h
index c5bac1f..7114bc4 100644
--- a/rtp.h
+++ b/rtp.h
@@ -60,8 +60,8 @@ public:
 	void stop();
 
 	int getHasLock() { return m_hasLock; }
-	int getSignalStrenth() { return m_signalStrength; }
-	int getSignalSNR() { return m_signalQuality; }
+	int getSignalStrength() { return m_signalStrength; }
+	int getSignalQuality() { return m_signalQuality; }
 };
 
 #endif
diff --git a/vtuner.cpp b/vtuner.cpp
index 1a21288..5291cb6 100644
--- a/vtuner.cpp
+++ b/vtuner.cpp
@@ -514,13 +514,13 @@ void satipVtuner::vtunerEvent()
 		case MSG_READ_SIGNAL_STRENGTH:
 			//INFO(MSG_MAIN,"MSG_READ_SIGNAL_STRENGTH %d -> %d\n", msg.body.ss, rtcp_data.signalStrength);
 			if (m_satip_rtp)
-				msg.body.ss = (u16)m_satip_rtp->getSignalStrenth();
+				msg.body.ss = (u16)m_satip_rtp->getSignalStrength();
 			break;
 
 		case MSG_READ_SNR:
 			//INFO(MSG_MAIN,"MSG_READ_SNR %d -> %d\n", msg.body.snr, m_satip_cfg->satip_get_signal_quality());
 			if (m_satip_rtp)
-				msg.body.snr = (u16)m_satip_rtp->getSignalSNR();
+				msg.body.snr = (u16)m_satip_rtp->getSignalStrength() * (u16)m_satip_rtp->getSignalQuality() / 65535;
 			break;
 
 		case MSG_READ_UCBLOCKS:
-- 
2.7.4

